### ISSUE TITLE
Dialog, DialogTitle a11y fixes not implemented

### ISSUE THREAD

**Issue by bluSCALE4** [jpmorganchase/salt-ds#283]
```Dialog``` should not read aloud all of this content. Old UITK applied ```...other``` which allowed users to override defaults such as ```aria-describedby``` which produces this issue. Similarly, ```DialogTitle``` forces h1 to be applied. A fix for this is to use ```disabledAria```. Ideally, we should be able to provide a heading value since we would still need to re-add the span wrapper to achieve the desired heading style.

```
    <Dialog
      aria-describedby={null}
    >
      <DialogTitle disableAria>
      </DialogTitle>
    </Dialog>
```

**Comment by joshwooding** [comment]
Hi @bluSCALE4, thanks for your report. We're still currently working on improving the components in labs so an accessibility review is on the list of things to do.

**Comment by joshwooding** [comment]
This was fixed by #2132
